src/System.IO.IsolatedStorage/src/System/IO/IsolatedStorage/IsolatedStorage.cs

```diff
@@ -0,0 +1,194 @@
+            }
+        }
+
+        [CLSCompliant(false)]
+        [Obsolete("IsolatedStorage.MaximumSize has been deprecated because it is not CLS Compliant.  To get the maximum size use IsolatedStorage.Quota")]
+        public virtual ulong MaximumSize
+        {
+            get
+            {
+                if (_validQuota)
+                    return _quota;
+
+                throw new InvalidOperationException(string.Format(SR.IsolatedStorage_QuotaIsUndefined, nameof(MaximumSize)));
+            }
+        }
+
+        public virtual long Quota
+        {
+            get
+            {
+                if (_validQuota)
+                    return (long)_quota;
+
+                throw new InvalidOperationException(string.Format(SR.IsolatedStorage_QuotaIsUndefined, nameof(Quota)));
+            }
+
+            internal set
+            {
+                _quota = (ulong)value;
+                _validQuota = true;
+            }
+        }
+
+        public IsolatedStorageScope Scope
+        {
+            get; protected set;
+        }
+
+        protected virtual char SeparatorExternal
+        {
+            get { return Path.DirectorySeparatorChar; }
+        }
+
+        protected virtual char SeparatorInternal
+        {
+            get { return '.'; }
+        }
+
+        public virtual bool IncreaseQuotaTo(long newQuotaSize)
+        {
+            return false;
+        }
+
+        public abstract void Remove();
+
+        protected string IdentityHash
+        {
+            get; private set;
+        }
+
+        protected void InitStore(IsolatedStorageScope scope, Type appEvidenceType)
+        {
+            InitStore(scope, null, appEvidenceType);
+        }
+
+        protected void InitStore(IsolatedStorageScope scope, Type domainEvidenceType, Type assemblyEvidenceType)
+        {
+            Scope = scope;
+
+            // NetFX (desktop CLR) IsolatedStorage uses identity from System.Security.Policy.Evidence to build
+            // the folder structure on disk. It would use the "best" available evidence in this order:
+            //
+            //  1. Publisher (Authenticode)
+            //  2. StrongName
+            //  3. Url (CodeBase)
+            //  4. Site
+            //  5. Zone
+            //
+            // For CoreFx StrongName and Url are the only relevant types. By default evidence for the Domain comes
+            // from the Assembly which comes from the EntryAssembly(). We'll emulate the legacy default behavior
+            // by pulling directly from EntryAssembly.
+
+            Assembly assembly = Assembly.GetEntryAssembly();
+            AssemblyName assemblyName = assembly.GetName();
+            Uri codeBase = new Uri(assembly.CodeBase);
+            object identity = null;
+
+            string hash = Helper.GetNormalizedStrongNameHash(assemblyName);
+            if (hash != null)
+            {
+                hash = $"StrongName{SeparatorInternal}{hash}";
```

Comment:
It'd be further better as:
```C#
hash = "StrongName" + SeparatorInternal.ToString() + hash;
```
Otherwise this will bind to the string.Concat(object, object, object) overload, resulting in the char in the middle getting boxed.